filters/auth/doc.go

```diff
@@ -1,18 +1,162 @@
-Each incoming request will be validated against the password file, for more information which formats are currently
-supported check "https://github.com/abbot/go-http-auth".
-Assuming you are going to use the MD5 version new entries can be generated like
+The filter accepts two parameters, the first mandatory one is the path to the htpasswd file usually used with
+Apache or nginx. The second one is the optional realm name that will be displayed in the browser. Each incoming
+request will be validated against the password file, for more information which formats are currently supported
+check "https://github.com/abbot/go-http-auth". Assuming that the MD5 version will be used, new entries can be generated like
 
 	htpasswd -nbm myName myPassword
 
-Usage
+Embedding the filter in routes:
+
+	generic: * -> basicAuth("/path/to/htpasswd") -> "https://internal.example.org";
+
+	myRealm: Host("my.example.org")
+	  -> basicAuth("/path/to/htpasswd", "My Website")
+	  -> "https://my-internal.example.org";
+
+Set Basic - Outgoing Authentication Header
 
 	basicAuth("/path/to/htpasswd")
 	basicAuth("/path/to/htpasswd", "My Website")
+
+OAuth2 - Check Bearer Tokens
+
+The auth filter takes the incoming request, and tries to extract the
+Bearer token from the Authorization header. Then it validates against
+a configured service. Depending on the settings, it also can check if
+the owner of the token belongs to a specific OAuth2 realm, and it can
+check if it has at least one of the predefined scopes. If any of the
+expectations are not met, it doesn't forward the request to the target
+endpoint, but returns with status 401.
+
+As additional feature, the package also supports audit logging.
+
+OAuth2 - Provider Configuration - Tokeninfo
+
+To enable OAuth2 tokeninfo filters you have to set the CLI argument
+-oauth2-tokeninfo-url=<OAuthTokeninfoURL>. Scopes and key value pairs
+depend on the OAuth2 tokeninfo provider. AccessTokens has to be
+accepted by your OAuth2 provider's TokeninfoURL. Filter names starting
+with oauthTokeninfo will work on the returned data from
+TokeninfoURL. The request from skipper to TokeninfoURL will use the
+query string to do the request:
+?access_token=<access-token-from-authorization-header>.
+
+Example json output of this tokeninfo could be:
+
+    {
+      "access_token": "<mytoken>",
+      "client_id": "ztoken",
+      "cn": "Jane Doe",
+      "expires_in": "300",
+      "grant_type": "password",
+      "realm": "/employees",
+      "scope": [
+        "uid",
+        "foo-r",
+        "bar-w",
+        "qux-rw"
+      ],
+      "token_type": "Bearer",
+      "uid": "jdoe"
+    }
+
+OAuth2 - oauthTokeninfoAnyScope filter
+
+The filter oauthTokeninfoAnyScope allows access if one of the scopes
+are satisfied by the request.
+
+    Path("/a") -> oauthTokeninfoAnyScope("uid") -> "https://internal.example.org/";
```

Comment:
when we have multiple routes in the same eskip doc separated by `;`, then the route id is mandatory. E.g:

```
a: Path("/a") -> oauthTokeninfoAnyScope("uid") -> "https://internal.example.org/";
b: Path("/b") -> oauthTokeninfoAnyScope("uid", "bar") -> "https://internal.example.org/";
```